python/pyarrow/_s3fs.pyx

```diff
@@ -94,6 +186,11 @@ cdef class S3FileSystem(FileSystem):
     background_writes: boolean, default True
         Whether OutputStream writes will be issued in the background, without
         blocking.
+    proxy_options: dict or pyarrow._s3fs.S3ProxyOptions, default None
+        If a proxy is used, provide the options here. Supported options are:
```

Comment:
Alternatively, we could also only accept a dict here, to avoid needing to create the public `S3ProxyOptions` class? (to limit the API exposure a bit)

In addition, it could also accept a string in which case it would do the equivalent of `S3ProxyOptions.from_uri(..)` ?

Reply:
+1 for also accepting a string.

Reply:
Thanks. That's a good idea. I'll implement this and push the changes shortly.

Reply:
Done. Let me know if this looks better.

Reply:
Thanks, that's better I think (nice reduction in lines of code ;))